src/utils/convert3Dto4D.m

```diff
@@ -0,0 +1,164 @@
+  %           - :argout2: (type) (dimension)
+  %
+  % .. todo:
+  %
+  %    - expand to run through multiple subjs ans groups
+  %      (https://stackoverflow.com/questions/8748976/list-the-subfolders-in-a-folder-matlab-only-subfolders-not-files)
+  %    - generalize how to retrieve RT from sidecar json file
+  %    - saveMatlabBatch(matlabbatch, ['3Dto4D_dataType-' num2str(dataType) '_RT-' num2str(RT)], opt, subID);
+
+  % Set the folder where sequences folders exist
+  optSource.sourceDir = '/Users/barilari/Desktop/DICOM_UCL_leuven/renamed/sub-pilot001/ses-002/MRI';
+
+  % List the sequences that you want to skip (folder name pattern)
+  optSource.sequenceToIgnore = {'AAHead_Scout', ...
+                                'b1map', ...
+                                't1', ...
+                                'gre_field'};
+
+  % Set data format conversion (0 is reccomended)
+
+  % 0:  SAME
+  % 2:  UINT8   - unsigned char
+  % 4:  INT16   - signed short
+  % 8:  INT32   - signed int
+  % 16: FLOAT32 - single prec. float
+  % 64: FLOAT64 - double prec. float
+  dataType = 0;
+
+  % Get source folder content
+  sourceDataStruc = dir(optSource.sourceDir);
+
+  isDir = [sourceDataStruc(:).isdir];
+
+  optSource.sequenceList = {sourceDataStruc(isDir).name}';
+
+  % Loop through the sequence folders
+
+  tic;
+
+  for iSeq = 1:size(optSource.sequenceList, 1)
+
+    % Skip 'non' folders
+    if length(char(optSource.sequenceList(iSeq))) > 2
+
+      % Check if sequence to ignore or not
+      if contains(optSource.sequenceList(iSeq), optSource.sequenceToIgnore)
+
+        newline;
+        warning('IGNORING SEQUENCE: %s', string(optSource.sequenceList(iSeq)));
+        newline;
+
+      else
+
+        fprintf('\n\nCONVERTING SEQUENCE: %s \n\n', char(optSource.sequenceList(iSeq)));
+
+        % Get sequence folder path
+        sequencePath = char(fullfile(optSource.sourceDir, optSource.sequenceList(iSeq)));
+
+        % Retrieve volume files info
+        [volumesList, outputNameImage] = parseFiles('nii', sequencePath);
+
+        % Set output name, it takes the file name of the first volume and add subfix '_4D'
+
+        dotPos = find(outputNameImage == '.');
+
+        outputNameImage = outputNameImage(1:dotPos(1) - 1);
+
+        outputNameImage = [outputNameImage '_4D.nii']; %#ok<AGROW>
+
+        % Retrieve sidecar json files info
+        [jsonList, outputNameJson] = parseFiles('json' ,sequencePath);
+
+        jsonFile = jsondecode(fileread(char(jsonList(1))));
+
+        % % % % % % LIEGE  SPECIFIC % % % % % % %
+        RT = jsonFile.acqpar.RepetitionTime / 1000;
+        % % % % % % % % % % % % % % % % % % % % %
+
+        % Set and run spm batch
+        matlabbatch = setBatch3Dto4D(volumesList, outputNameImage, dataType, RT);
+
+        spm_jobman('run', matlabbatch);
+
+        % Zip and delete the and the new 4D file
+        fprintf(1, 'ZIP AND DELETE THE NEW 4D BRAIN \n\n');
+
+        gzip([sequencePath filesep outputNameImage]);
```

Comment:
Remind me to factor out the unzipping part of the `bidsCopyRaw...` because it seems it could be reused in places. :wink: